dataprofiler/profilers/profile_builder.py

```diff
@@ -1857,6 +1864,47 @@ def _merge_correlation_helper(corr_mat1, mean1, std1, n1,
 
         return corr_mat
 
+    def _update_chi2(self):
+        n_cols = len(self._profile)
+        cat_cols = {}
+        cat_col_idxs = []
+
+        # Get categorical columns
+        for i in range(n_cols):
+            compiler = self._profile[i]
+            data_stats_compiler = compiler.profiles["data_stats_profile"]
+            profiler = data_stats_compiler._profiles["category"]
+            if profiler.is_match:
+                categories = profiler._categories
```

Comment:
let's not use the private variable here and just use `cateogries`

Reply:
I use _categories to get the counts. Any other ways to do this?